### PR TITLE
[SELC-5173] feat: API invocation for private body search via CF

### PR DESCRIPTION
**Pull request by GabrieleMaiocchiFilippo**
#### List of Changes

[SELC-5173] feat: API invocation for private body search via CF

#### Motivation and Context

As a user of an organization who wants to subscribe to a product, I want to be able to select my organization by searching for it by tax code on the Infocamere data source

#### How Has This Been Tested?

tested if inserting a taxCode the API on the infocamere data source retrives the wanted data

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

[SELC-5173]: https://pagopa.atlassian.net/browse/SELC-5173?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/autocomplete/components/asyncAutocomplete/AsyncAutocompleteContainer.tsx [hunk 1]
```diff
@@ -388,8 +387,19 @@ export default function AsyncAutocompleteContainer({
 
           void contractingInsuranceFromTaxId(addUser, endpoint, params, value);
         } else {
-          const endpoint = addUser ? 'ONBOARDING_GET_INSTITUTIONS' : 'ONBOARDING_GET_PARTY_FROM_CF';
-          void handleSearchByTaxCode(addUser, endpoint, params, value);
+          const endpoint = addUser
+            ? 'ONBOARDING_GET_INSTITUTIONS'
+            : product?.id === 'prod-interop'
```
Comment (loremedda) [inline src/components/autocomplete/components/asyncAutocomplete/AsyncAutocompleteContainer.tsx:392]: By doing this, the search by tax code on infocamere will always be valid

File: src/components/autocomplete/components/asyncAutocomplete/components/AsyncAutocompleteResultsCode.tsx [hunk 2]
```diff
@@ -84,7 +84,7 @@ export default function AsyncAutocompleteResultsCode({
                 ? aooResult?.denominazioneAoo.toLocaleLowerCase()
                 : isUoCodeSelected && uoResult?.descrizioneUo
                 ? uoResult?.descrizioneUo.toLocaleLowerCase()
-                : visibleCode?.description
+                : (cfResult as any).businessName.toLocaleLowerCase()
```
Comment (loremedda) [inline src/components/autocomplete/components/asyncAutocomplete/components/AsyncAutocompleteResultsCode.tsx:87]: Why lowerCase? You should have already the businessName in lower case with first letter capitalized

File: src/components/autocomplete/components/partyAdvancedSearchType/PartyAdvancedSelect.tsx [hunk 3]
```diff
@@ -97,11 +104,13 @@ export default function PartyAdvancedSelect({
...
     } else if (isIvassCodeSelected) {
       setTypeOfSearch('ivassCode');
+    } else {
+      setTypeOfSearch('');
     }
   }, []);
 
   useEffect(() => {
-    if (addUser) {
+    if (addUser || institutionType === 'SCP') {
```
Comment (loremedda) [inline src/components/autocomplete/components/partyAdvancedSearchType/PartyAdvancedSelect.tsx:113]: It may introduce regressions and unexpected behavior

**Review by loremedda — CHANGES_REQUESTED** [review]

File: src/components/autocomplete/components/asyncAutocomplete/AsyncAutocompleteContainer.tsx (hunk 1 above)
Comment (GabrieleMaiocchiFilippo) [reply src/components/autocomplete/components/asyncAutocomplete/AsyncAutocompleteContainer.tsx]: added secondary condition to search by tax code on infocamere only for a specific case

File: src/components/autocomplete/components/asyncAutocomplete/components/AsyncAutocompleteResultsCode.tsx (hunk 2 above)
Comment (GabrieleMaiocchiFilippo) [reply src/components/autocomplete/components/asyncAutocomplete/components/AsyncAutocompleteResultsCode.tsx]: deleted lowercase on businessName

**Comment by loremedda** [comment]
OK

**Review by andrea-putzu — APPROVED** [review]

**Review by loremedda — APPROVED** [review]

**Pull request merged**